packages/layout/src/footnotes/chooseFootnotes.js

```diff
@@ -0,0 +1,88 @@
+import * as R from 'ramda';
+import getFootnotePlaceholder from './getFootnotePlaceholder';
+
+const getStyleHeight = R.pathOr(0, ['style', 'height']);
+const getHeight = R.path(['box', 'height']);
+
+const FOOTNOTES_MIN_HEIGHT = 15;
+
+const groupByKey = (array, key) => {
+  return Object.values(
+    array.reduce((r, a) => {
+      r[a[key]] = r[a[key]] || [];
```

Comment:
Can we re-write this _groupByKey_ functionality in a way that does not modify function parameters?